python_modules/libraries/dagster-dbt/dagster_dbt/dbt_project.py

```diff
@@ -158,6 +174,8 @@ def __init__(
         target: Optional[str] = None,
         packaged_project_dir: Optional[Union[Path, str]] = None,
         state_path: Optional[Union[Path, str]] = None,
+        dependencies_path: Union[Path, str] = Path("dependencies.yml"),
+        packages_install_path: Union[Path, str] = Path("dbt_packages"),
```

Comment:
https://docs.getdbt.com/docs/build/packages

We shouldn't need to parameterize these paths. It will always be `dependencies.yml` and `dbt_packages` at the top level.

Reply:
I think it's best practice, but not mandatory:

- `dbt_packages` can be changed to another path via `packages-install-path` in `dbt_project.yml`, see [here](https://docs.getdbt.com/reference/project-configs/packages-install-path)
- `dependencies.yml` should be at top level and named this way, but there's no guarantee, see [here](https://docs.getdbt.com/docs/build/packages#how-do-i-add-a-package-to-my-project)
    - "_This should be at the same level as your dbt_project.yml file._" - but must not.
        - I tested it to make sure - `dependencies.yml` can be in a subfolder and `dbt deps` succeeds without any exception.
    - "_Add a file named dependencies.yml or packages.yml to your dbt project._" - there could be a `packages.yml` file, but no `dependencies.yml`.

We could remove the parameters for paths, but we should make it clear in the docs that `DbtProject` supports dbt projects that follow best practice. Though, I'm not sure the experience is optimal for users.

Reply:
> Add a file named dependencies.yml or packages.yml to your dbt project.

We can just check for the presence of both.

> `dbt_packages` can be changed to another path via packages-install-path in `dbt_project.yml`

Sure, but if that's the case, then we can just read `dbt_project.yml` and figure out the path from that. We still won't need to parameterize the path since we can figure it out on behalf of the user.